From 0351b551fd26891e23e39147a9040a9aa3ddc448 Mon Sep 17 00:00:00 2001
From: Danny Kukawka <danny.kukawka@web.de>
Date: Fri, 9 Apr 2010 12:40:50 +0200
Subject: [PATCH 23/33] added adopted version of a SUSE patch to support HP keymaps

Added adopted version of a SUSE patch to support HP keymaps from
thoenig@suse.de and hmacht@suse.de. (see bnc#550670, bnc#552532
and bnc#577998)
---
 fdi/information/10freedesktop/30-keymap-hp.fdi |   17 +++++++++++++++++
 1 files changed, 17 insertions(+), 0 deletions(-)

diff --git a/fdi/information/10freedesktop/30-keymap-hp.fdi b/fdi/information/10freedesktop/30-keymap-hp.fdi
index c49cc47..c230a2a 100644
--- a/fdi/information/10freedesktop/30-keymap-hp.fdi
+++ b/fdi/information/10freedesktop/30-keymap-hp.fdi
@@ -20,6 +20,23 @@
           <append key="input.keymap.data" type="strlist">e057:wlan</append>     <!-- Wlan -->
 	  <append key="input.keymap.data" type="strlist">e06e:switchvideomode</append>   <!-- FnF4 (video switch LCD/CRT) -->
           <append key="info.capabilities" type="strlist">input.keymap</append>
+	  
+	  <!-- We're so sorry... -->
+ 	  <match key="/org/freedesktop/Hal/devices/computer:system.firmware.release_date" contains_outof="2009;2010;2011;2012;2013;2014;2015">
+            <!-- [make code:kernel define lower case (linux/input.h)]                    comment: description according to HP spec -->
+            <append key="input.keymap.data" type="strlist">e008:presentation</append>    <!-- presentation mode -->
+            <append key="input.keymap.data" type="strlist">e012:brightnessdown</append>  <!-- brightness down -->
+            <append key="input.keymap.data" type="strlist">e017:brightnessup</append>    <!-- brightness up -->
+            <append key="input.keymap.data" type="strlist">e020:mute</append>            <!-- volume mute -->
+            <append key="input.keymap.data" type="strlist">e02e:volumedown</append>      <!-- volume down -->
+            <append key="input.keymap.data" type="strlist">e030:volumeup</append>        <!-- volume up  -->
+            <append key="input.keymap.data" type="strlist">e05f:sleep</append>           <!-- sleep -->
+            <append key="input.keymap.data" type="strlist">e06b:computer</append>        <!-- browser button (cf. bnc#550670 comment 17) -->
+            <append key="input.keymap.data" type="strlist">e06c:mail</append>	          <!-- envelope button (cf. bnc#550670 comment 17) -->
+            <append key="input.keymap.data" type="strlist">e06e:switchvideomode</append> <!-- LCD/CRT -->
+            <append key="input.keymap.data" type="strlist">e075:info</append>            <!-- info center -->
+            <append key="input.keymap.data" type="strlist">e078:wlan</append>            <!-- wireless toggle -->
+	  </match>
 
 	  <!-- We're so sorry... -->
           <match key="/org/freedesktop/Hal/devices/computer:system.firmware.release_date" contains_outof="2009;2010;2011;2012;2013;2014;2015">
-- 
1.7.1

